python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -1011,7 +1018,33 @@ def _transform_arg(
         # use __reduce_ex__ instead...
         if type(arg) is tuple:
             # Must come first to avoid infinite recursion
-            return tuple(_transform_arg(a, attribute_name, seen) for a in arg)
+            if (
+                len(arg) > 0
+                and isinstance(arg[0], _MethodProxy)
```

Comment:
Why `_MethodProxy`? Is `pd.Index` pass that check?

Reply:
That's for `__setitem__`?

Reply:
`DataFrame.__seitem__` and other functions are wrapped into `_MethodProxy`